activerecord/test/cases/adapters/postgresql/array_test.rb

```diff
@@ -377,6 +378,27 @@ def test_precision_is_respected_on_timestamp_columns
     assert_equal 123, record.timestamps.first.usec
   end
 
+  def test_tsrange_with_timezone_awareness
```

Comment:
What do you think about moving this test to https://github.com/rails/rails/blob/main/activerecord/test/cases/adapters/postgresql/range_test.rb ?

Reply:
@yahonda done.

BTW, I found that when I set `PostgresqlRange.new(ts_ranges: [time_string...time_string])`, so the same time for exclusive range, it silently sets nil on it. Do you know if this is expected behaviour?

Reply:
```ruby
record = PostgresqlRange.new(ts_ranges: [from_time_string...from_time_string, from_time_string..from_time_string])
record.ts_ranges
=> [Tue, 21 Jun 2022 06:51:36.000000000 PDT -07:00...Tue, 21 Jun 2022 06:51:36.000000000 PDT -07:00, Tue, 21 Jun 2022 06:51:36.000000000 PDT -07:00..Tue, 21 Jun 2022 06:51:36.000000000 PDT -07:00]
record.save
=> true
record.ts_ranges
=> [Tue, 21 Jun 2022 06:51:36.000000000 PDT -07:00...Tue, 21 Jun 2022 06:51:36.000000000 PDT -07:00, Tue, 21 Jun 2022 06:51:36.000000000 PDT -07:00..Tue, 21 Jun 2022 06:51:36.000000000 PDT -07:00]
record.reload.ts_ranges
=> [nil, Tue, 21 Jun 2022 06:51:36.000000000 PDT -07:00..Tue, 21 Jun 2022 06:51:36.000000000 PDT -07:00]
```

Anyway - this is not related to this issue.

Reply:
I think this behavior comes from the PostgreSQL implementation. It reproduces with the plain sql statements below. The duplicate entry becomes `empty`.

```
postgres=# select version();
                                                       version
---------------------------------------------------------------------------------------------------------------------
 PostgreSQL 14.4 on aarch64-apple-darwin21.5.0, compiled by Apple clang version 13.1.6 (clang-1316.0.21.2.5), 64-bit
(1 row)
postgres=# create table ranges_test(ts_ranges tsrange[]);
INSERT INTO ranges_test(ts_ranges) VALUES ('{"[2022-06-27 11:00:00,2022-06-27 11:00:00)","[2022-06-27 11:00:00,2022-06-27 11:00:00]"}');
select * from ranges_test;
CREATE TABLE
INSERT 0 1
                          ts_ranges
-------------------------------------------------------------
 {empty,"[\"2022-06-27 11:00:00\",\"2022-06-27 11:00:00\"]"}
(1 row)

postgres=#
```